homeassistant/components/tibber/sensor.py

```diff
@@ -45,108 +49,142 @@
 PARALLEL_UPDATES = 0
 SIGNAL_UPDATE_ENTITY = "tibber_rt_update_{}"
 
-RT_SENSOR_MAP = {
-    "averagePower": ["average power", DEVICE_CLASS_POWER, POWER_WATT, None],
-    "power": ["power", DEVICE_CLASS_POWER, POWER_WATT, None],
-    "powerProduction": ["power production", DEVICE_CLASS_POWER, POWER_WATT, None],
-    "minPower": ["min power", DEVICE_CLASS_POWER, POWER_WATT, None],
-    "maxPower": ["max power", DEVICE_CLASS_POWER, POWER_WATT, None],
-    "accumulatedConsumption": [
+
+class ResetType(Enum):
+    """Data reset type."""
+
+    HOURLY = "hourly"
+    DAILY = "daily"
+    NEVER = "never"
+
+
+@dataclass
+class TibberSensorMetadata:
```

Comment:
Why not use a `NamedTuple` (from `typing`) for it? As far as I can tell the values don't change after the initialization. You also didn't add any additional methods to the class.

Reply:
Thanks for your review. 
Could you explain why NamedTuple is better than dataclass?

Reply:
`NamedTuple` are immutable and thus a bit more efficient. I usually prefer them if I only need to store some basic information which isn't going to change. A `dataclass` in contrast is really useful if you need to store data (hence the name) which is going to change and where you need some additional functionality. E.g., they add default compare methods, you can add a `__post_init__` hook and even custom methods. They are basically normal classes with something extra.

An example use of a `dataclass`: https://github.com/home-assistant/core/blob/9b705ad6dfddaa97ea135b93ce570a284c5ae731/homeassistant/components/climacell/const.py#L153-L173

Reply:
Thanks for the explanation. Will fix.